public/app/features/explore/RichHistory/RichHistoryCard.tsx

```diff
@@ -107,9 +107,17 @@ export function RichHistoryCard(props: Props) {
     }
   };
 
+  useEffect(() => {
+    setStared(query.starred);
+  }, [query.starred]);
+
+  useEffect(() => {
+    setComment(query.comment);
+  }, [query.comment]);
```

Comment:
This feels weird. If we are changing the comment in this component who would change the comment outside of the component?

Reply:
yep, weird. I have super over-complicated it. I just looked at it in a fresh perspective and it is fixed.

Reply:
> This feels weird. If we are changing the comment in this component who would change the comment outside of the component?

yep, weird. I have super over-complicated it. I just looked at it in a fresh perspective and it is fixed.

Reply:
>This seems weird is it a bug in the form elements?

No, it is expected behaviour. You cave a clickable card - when you click on the card, it adds query to query editor (we might change this behaviour and instead of clickable card we add the button). And on that card we add buttons and textarea. So if you click on them, it bubbles up and also runs the function to add query to query editor. So we need to add stop propagation.

Reply:
>I would still like to lift this one level up. So I would rather have this function throw the error, handle it in the both the RichHistory helpers and in LocalStorageValueProvider where this is used. That way LocalStorageValueProvider can still keep the console.log as we don't know if the emit is appropriate there, and add the emit to rich history where we know we want it. We can do it in separate PR.

I have removed it from this PR, I will open then a new PR with just error handling of store.